x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -187,3 +207,20 @@ func newJSONDecoder(r io.Reader) *json.Decoder {
 	dec.UseNumber()
 	return dec
 }
+
+// getBodyReader returns a reader that decodes the specified Content-Encoding.
+func getBodyReader(r *http.Request) (body io.ReadCloser, status int, err error) {
+	switch enc := r.Header.Get(headerContentEncoding); enc {
+	case "gzip":
```

Comment:
Is it worth supporting x-gzip, or should we drag people in to the modern age?

Reply:
Sounds like a good idea. To back that up https://datatracker.ietf.org/doc/html/rfc6648#appendix-B:

> For compatibility with previous implementations of HTTP, applications should consider "x-gzip" and "x-compress" to be equivalent to "gzip" and "compress" respectively.